surefire-providers/surefire-junit-platform/src/test/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProviderTest.java

```diff
@@ -410,11 +410,19 @@ public void allDiscoveredTestsAreInvokedForNullArgument()
         InOrder inOrder = inOrder( runListener );
 
         ArgumentCaptor<SimpleReportEntry> report = ArgumentCaptor.forClass( SimpleReportEntry.class );
-        inOrder.verify( runListener )
+        inOrder.verify( runListener, times( 2 ) )
```

Comment:
What was the second call of `testSetStarting`?
I just want to understand it.

Reply:
The engine container.
In theory one testplan can contain both the normal Jupiter engine and also the "Vintage" engine (which will execute Junit4 tests). Also more engines can be written by users.